test/Microsoft.AspNet.Mvc.FunctionalTests/ViewEngineTests.cs

```diff
@@ -289,5 +305,80 @@ public static IEnumerable<object[]> RazorViewEngine_UsesExpandersForLayoutsData
             // Assert
             Assert.Equal(expected, body.Trim());
         }
+
+        [Fact]
+        public async Task ViewComponentsExecuteLayout()
+        {
+            // Arrange
+            var expected =
+@"<title>View With Component With Layout</title>
+
+Page Content
+<component-title>ViewComponent With Title</component-title>
+<component-body>
+Component With Layout</component-body>";
+            var server = TestServer.Create(_provider, _app);
+            var client = server.CreateClient();
+
+            // Act
+            var body = await client.GetStringAsync("http://localhost/ViewEngine/ViewWithComponentThatHasLayout");
+
+            // Assert
+            Assert.Equal(expected, body.Trim());
+        }
+
+        [Fact]
+        public async Task PartialDoNotExecuteViewStarts()
+        {
+            // Arrange
+            var expected = "Partial that does not specify Layout";
+            var server = TestServer.Create(_provider, _app);
+            var client = server.CreateClient();
+
+            // Act
+            var body = await client.GetStringAsync("http://localhost/PartialsWithLayout/PartialDoesNotExecuteViewStarts");
+
+            // Assert
+            Assert.Equal(expected, body.Trim());
```

Comment:
View components are executed as partial views and consequently have the same-ish behaviors as `PartialResult` \ `RenderPartial` \ `PartialAsync`. I'll go ahead and add a `_ViewStart` to the current example to cover ViewComponents.